### PR TITLE
Bug 1983220: Fix pod terminal second scrollbar when user reduce the window size

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5630

**Analysis / Root cause**: 
Xterm uses an invisible textarea to detect keyboard events. This textarea has an absolute position and was permanently moved when the cursor position is changed. But this does not happen when the user resizes the window.

Bad positioned textarea:

textarea-position

**Solution Description**: 
This was NOT solved by updating xterm to 4.13.0. Hoped that because I found this issue: https://github.com/xtermjs/xterm.js/pull/3169

Open a new xterm issue: https://github.com/xtermjs/xterm.js/issues/3390

As workaround call the internal `_syncTextArea` when the window was resized. This method was currently only called when the cursor moved: https://github.com/xtermjs/xterm.js/blob/4.10.0/src/browser/Terminal.ts#L474-L477

**Screen shots / Gifs for design review**: 

Broken before in Chrome:

broken-chrome

Broken before in Firefox:

broken-firefox

Fixed in Chrome:

fixed-chrome

Fixed in Firefox:

fixed-firefox

**Unit test coverage report**: 
Unchanged.

**Test setup:**
1. Create a Deployment
2. Open Pod detail page and switch to "Terminal" tab
3. Create a long output, for example via `find .` or `find /`
4. Reduce the window height

**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by jerolimov** [comment]
/cc @sahil143
Can you review this change? Thanks :)

File: frontend/public/components/terminal.jsx [hunk 1]
```diff
@@ -116,8 +115,11 @@ class Terminal_ extends React.Component {
       this.fitAddon.fit();
       // update the pty
       this.props.onResize(terminal.rows, terminal.cols);
+      // The internal xterm textarea was not repositioned when the window was resized.
+      // This workaround triggers an position update.
+      this.terminal._core?._syncTextArea?.();
```
Comment (TheRealJon) [inline frontend/public/components/terminal.jsx:120]: Correct me if I'm wrong, but this seems to be the only line needed to fix the bug. I would prefer that the rest of the changes (besides your comments here) be removed from the PR as they don't seem to be relevant to this bug fix.

**Review by TheRealJon — CHANGES_REQUESTED** [review]
@jerolimov I opened a bugzilla bug against OCP Management Console to make merging this easier, but see my comments about narrowing down the changes in this PR.

File: frontend/public/components/terminal.jsx (hunk 1 above)
Comment (jerolimov) [reply frontend/public/components/terminal.jsx]: @TheRealJon your right, and it also make sense to keep this PR small. I dropped all other changes and retested the fix, still works fine. PTAL.

**Review by TheRealJon — APPROVED** [review]
/lgtm
/approve
